src/Kuzzle.js

```diff
@@ -443,28 +443,34 @@ Discarded request: ${JSON.stringify(request)}`));
   /**
    * Adds a new controller and make it available in the SDK.
    *
-   * @param {BaseController} controller
+   * @param {BaseController} ControllerClass
+   * @param {string} accessor
    * @returns {Kuzzle}
    */
-  useController (controller) {
-    if (!(controller instanceof BaseController)) {
+  useController (ControllerClass, accessor) {
+    if (!(ControllerClass.prototype instanceof BaseController)) {
       throw new Error('Controllers must inherits from the BaseController class.');
     }
 
-    if (!(controller.name && controller.name.length > 0)) {
-      throw new Error('Controllers must have a name.');
+    if (!(accessor && accessor.length > 0)) {
+      throw new Error('You must provide a valid accessor.');
+    }
+
+    if (this[accessor]) {
+      throw new Error(`There is already a controller with the accessor '${accessor}'. Please use another one.`);
     }
 
-    if (!(controller.accessor && controller.accessor.length > 0)) {
-      throw new Error('Controllers must have an accessor.');
+    const controller = new ControllerClass(this);
+
+    if (!(controller.name && controller.name.length > 0)) {
```

Comment:
Didn't you mean
```suggestion
    if (!(controller.name && controller.name.length <= 0)) {
```
?

Reply:
No ^^

Reply:
I was just testing you !